### DISCUSSION TITLE
Detector offset x-axis

### DISCUSSION THREAD

**Discussion by rodrigovimieiro** [CERN/TIGRE#456] [Q&A] [answered]
Is there a way to apply an Offset in the detector on the X-axis? That would be similar to changing the geo.DSD per angle.

I did realize there is a way to apply on Z and Y. Also, it is possible to apply o the origin for all axes. However, I am interested in applying the X-axis for the detector itself, particularly for negative values, i.e., increasing the DSD. 

Thanks!

**Comment by AnderBiguri** [comment, marked as answer]
Hi Rodrigo,

Well, the way you do it in TIGRE is by increasing the DSD, which can be given per-projection. 

Ander

**Reply by rodrigovimieiro** [reply]
Hi Ander,

Are the offOrigin and offDetector considered before or after the tube/detector rotation?

In other words, should I calculate them considering at the 0° or after rotating, e.g., after a 90° rotation?

I am asking that because DSD, even though it could be set per projection, it is calculated relative to the 0° position, right?

Thanks!

**Reply by AnderBiguri** [reply]
@rodrigovimieiro offOrigin is at 0 deg indeed, you can imagine it as a detector rotating and an image being static. The offset of the image is irrelevant w.r.t. detector. 

offDetector (and rotDetector) are calculated after the angles, "in place", if that makes sense. They are constant w.r.t. the detector reference system. A small shift of the detector, for all angles, is just `offDetector=val`, same with the rotation. 

DSD and DSO are a distance, not a vector, so again, the angles should not have an effect. 

I can't really answer your question without knowing what exactly you want to simulate, though :)

**Reply by rodrigovimieiro** [reply]
I am trying to implement a calibrated geometry for CT, i.e., I have small shifts in X, Y and Z for the tube and for the detector for each angle. I can calculate those shifts at 0 deg or at each angle, if that makes sense.

**Reply by AnderBiguri** [reply]
Basically it all depends on what you mean by X,Y,Z, i.e. the answer depends on what reference system you are using. Say your calibration says that your detector, for any arbitrary angle, is slightly closer to the tube than what you though, always, by `val`. 

In TIGRE, you just need to change `geo.DSD=geo.DSD-val` for this case. But depends what your method outputs, you may need to tranform the data to "TIGRE friendly" or not. If your method gives you `[-val,0,0]` for angle=0, but `[0,val,0]` for angle=90deg, then you will need to transform your data. If your method gives you the offset w.r.t. the detector reference system, then it would give you always `[0,0,-val]` (for example, for UVW reference), so no need to translate. 

In short, it depends on how you have the data.
